src/jit/codegenxarch.cpp

```diff
@@ -7890,7 +7890,89 @@ CodeGen::getBaseVarForPutArgStk(GenTreePtr treeNode)
     {
         // The first baseVarNum is guaranteed to be the first incoming arg of the method being compiled.
         // See lvaInitTypeRef() for the order in which lvaTable entries are initialized.
+        // For Windows the value is always 0 (there are always slots in the caller's frame for the first 4 args to a callee).
+        // For System V the first stack arg should be calculated.
         baseVarNum = 0;
+
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+        CORINFO_ARG_LIST_HANDLE argLst = compiler->info.compMethodInfo->args.args;
+        const unsigned argSigLen = compiler->info.compMethodInfo->args.numArgs;
+        
+        unsigned usedIntRegs = 0;
+        unsigned usedFloatRegs = 0;
+        
```

Comment:
There could be multiple fast tail calls within a method.  We would end up computing first stack arg for each call unecessarily